core/src/components/modal/modal.tsx

```diff
@@ -139,19 +156,22 @@ export class TdsModal {
           } `}
         >
           <div class="tds-modal-header">
-            <slot name="tds-modal-headline"></slot>
+            {this.header && <div class="tds-modal-header-title">{this.header}</div>}
+            {usesHeaderSlot && <slot name="header"></slot>}
             <button class="tds-modal-close" aria-label="close" onClick={() => this.handleClose()}>
               <tds-icon name="cross" size="20px"></tds-icon>
             </button>
           </div>
 
           <div class="tds-modal-body">
-            <slot name="tds-modal-body"></slot>
+            <slot name="body"></slot>
           </div>
 
-          <div class="tds-modal-actions">
-            <slot name="tds-modal-actions"></slot>
-          </div>
+          {usesActionsSlot && (
+            <div class="tds-modal-actions">
+              <slot name="actions"></slot>
+            </div>
```

Comment:
Could this one be? 

```suggestion
              <slot name="actions"></slot>
```

Reply:
I think that would work, good suggestion, might need to tweak CSS a bit, but not a big problem.